src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs

```diff
@@ -275,44 +275,53 @@ static string SaveRandomConfig(string appSettingsPath)
         [ActiveIssue("https://github.com/dotnet/runtime/issues/34582", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
         public async Task CreateDefaultBuilder_SecretsDoesReload()
         {
-            var secretId = Assembly.GetExecutingAssembly().GetName().Name;
-            var reloadFlagConfig = new Dictionary<string, string>() { { "hostbuilder:reloadConfigOnChange", "true" } };
-            var secretPath = PathHelper.GetSecretsPathFromSecretsId(secretId);
-            var secretFileInfo = new FileInfo(secretPath);
-
-            Directory.CreateDirectory(secretFileInfo.Directory.FullName);
-
-            static string SaveRandomSecret(string secretPath)
+            var originalValue = Environment.GetEnvironmentVariable("DOTNET_USE_POLLING_FILE_WATCHER");
+            try
             {
-                var newMessage = $"Hello ASP.NET Core: {Guid.NewGuid():N}";
-                File.WriteAllText(secretPath, $"{{ \"Hello\": \"{newMessage}\" }}");
-                return newMessage;
-            }
+                Environment.SetEnvironmentVariable("DOTNET_USE_POLLING_FILE_WATCHER", "1");
```

Comment:
Do these tests run concurrently?  If so, then setting the environment variable for the entire process might effect other tests, you can use RemoteExecutor to give the test it's own process, or use API to setup the polling file-watcher.
https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/Microsoft.Extensions.FileProviders.Physical/tests/PhysicalFileProviderTests.cs#L117-L118